server/tserver/src/main/java/org/apache/accumulo/tserver/logger/LogReader.java

```diff
@@ -123,36 +122,17 @@ public void execute(String[] args) throws Exception {
       Set<Integer> tabletIds = new HashSet<>();
 
       for (String file : opts.files) {
-
         Path path = new Path(file);
-        LogFileKey key = new LogFileKey();
-        LogFileValue value = new LogFileValue();
-
-        if (fs.getFileStatus(path).isFile()) {
-          // read log entries from a simple hdfs file
-          try (final FSDataInputStream fsinput = fs.open(path);
-              DataInputStream input = DfsLogger.getDecryptingStream(fsinput, siteConfig)) {
-            while (true) {
-              try {
-                key.readFields(input);
-                value.readFields(input);
-              } catch (EOFException ex) {
-                break;
-              }
-              printLogEvent(key, value, row, rowMatcher, ke, tabletIds, opts.maxMutations);
-            }
-          } catch (LogHeaderIncompleteException e) {
-            log.warn("Could not read header for {} . Ignoring...", path);
-            continue;
-          }
-        } else {
-          // read the log entries sorted in a map file
-          try (RecoveryLogReader input = new RecoveryLogReader(fs, path)) {
-            while (input.hasNext()) {
-              Entry<LogFileKey,LogFileValue> entry = input.next();
-              printLogEvent(entry.getKey(), entry.getValue(), row, rowMatcher, ke, tabletIds,
-                  opts.maxMutations);
-            }
+        if (!fs.getFileStatus(path).isDirectory()) {
+          log.error("No directory was given. Please pass in a recovery directory");
+          continue;
+        }
+        // read the log entries sorted in a RFile
+        try (var rli = new RecoveryLogsIterator(context, Collections.singletonList(path), true)) {
+          while (rli.hasNext()) {
+            Entry<LogFileKey,LogFileValue> entry = rli.next();
+            printLogEvent(entry.getKey(), entry.getValue(), row, rowMatcher, ke, tabletIds,
+                opts.maxMutations);
```

Comment:
This looks good for reading the sorted logs but we still want the section under `if (fs.getFileStatus(path).isFile())` for reading regular write ahead logs using `DfsLogger`.

Reply:
My mistake, I thought we previously discussed removing it. I will add it back.

Reply:
Also, if a user specifies a single sorted WAL to read, it will error out (the second part of your issue mentions this). I will look into adding a check to prevent this though a more descriptive error message could suffice.

Reply:
OK cool. If user passes in an rfile I think you could just create a singleton list to send as the param for `recoveryLogDirs`.

Reply:
A single rfile will not work with RLI currently. It doesn't contain the finish marker which is looked for in rli.getFiles (that seems to be the only issue with it not working). Also, since LogReader usually deals with one directory/file at a time since it's looped through what the user passes in, I already am using a singleton list to read in that path. I could change the passed in files (opts.files in LogReader) to be a List<path> but it doesn't seem necessary.

Reply:
Gotcha. I think making the user pass in the directory is fine. It should print the error if it doesn't have the finished file, correct? I think reading a partially sorted directory is undesirable so that should work.

Reply:
My latest commit adds an extra error message if it detects a file but the file ends with .rf. Otherwise, yes, it will error out if it gets past that but doesn't contain the finished file.